src/extension/ogcapi/ogcapi-core/src/main/java/org/geoserver/ogcapi/TimeExtentCalculator.java

```diff
@@ -68,19 +70,27 @@ private static DateRange getTimeExtent(FeatureTypeInfo ft, DimensionInfo time) t
         // check calcresult first to avoid potential IllegalStateException if no features are in
         // collection
         if (minResult != CalcResult.NULL_RESULT) {
-            Date minDate = (Date) min.getMin();
+            Date minDate = getDate(min.getMin());
             String endTimeAttribute =
                     Optional.ofNullable(time.getEndAttribute()).orElse(timeAttribute);
             final MaxVisitor max = new MaxVisitor(endTimeAttribute);
             collection.accepts(max, null);
-            Date maxDate = (Date) max.getMax();
-
+            Date maxDate = getDate(max.getMax());
             if (minDate != null && maxDate != null) return new DateRange(minDate, maxDate);
         }
 
         return null;
     }
 
+    public static Date getDate(Comparable timeValue) {
```

Comment:
Is it really needed, or a case of double fix? Maybe it's done because min/max/unique are not converted? 
See if the fix in the geoparquetDialect also allows to remove the change here.
Again for context, if the feature type says a property is Date.class, so should be its min/max value.